src/network/networkd-json.c

```diff
@@ -1170,6 +1171,31 @@ static int dhcp_client_pd_append_json(Link *link, JsonVariant **v) {
         return json_variant_set_field_non_null(v, "6rdPrefix", array);
 }
 
+static int dhcp_client_private_options_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp_lease)
+                return 0;
+
+        LIST_FOREACH(options, option, link->dhcp_lease->private_options) {
+                char key[STRLEN("Option_000")+1];
+
+                xsprintf(key, "OPTION_%" PRIu8, option->tag);
+
+                r = json_variant_append_arrayb(
+                                &array,
+                                JSON_BUILD_OBJECT(
+                                               JSON_BUILD_PAIR_HEX(key, option->data, option->length)));
+                if (r < 0)
+                        return 0;
+        }
+        return json_variant_set_field_non_null(v, "PrivateOptions", array);
```

Comment:
These are called in the other settings "VendorOptions", so better to keep it consistent

Reply:
vendor option is option code 42, but private options here is 224-254. So, I think the name `PrivateOptions` is fine.

Reply:
ok